src/software/backend/simulation/physics/physics_robot.cpp

```diff
@@ -24,119 +63,406 @@ PhysicsRobot::~PhysicsRobot()
     }
 }
 
-void PhysicsRobot::createRobotPhysicsBody(std::shared_ptr<b2World> world,
-                                          const Robot& robot)
+void PhysicsRobot::setupRobotBodyFixtures(const Robot& robot, double total_chicker_depth,
+                                          double mass_kg)
 {
-    // All the BodyDef must be defined before the body is created.
-    // Changes made after aren't reflected
-    robot_body_def.type = b2_dynamicBody;
-    robot_body_def.position.Set(robot.position().x(), robot.position().y());
-    robot_body_def.linearVelocity.Set(robot.velocity().x(), robot.velocity().y());
-    robot_body_def.angle           = robot.orientation().toRadians();
-    robot_body_def.angularVelocity = robot.angularVelocity().toRadians();
+    b2FixtureDef robot_body_fixture_def;
+    // This is a somewhat arbitrary value. Collisions with robots are not perfectly
+    // elastic. However because this is an "ideal" simulation and we generally don't care
+    // about the exact behaviour of collisions, getting this value to perfectly match
+    // reality isn't too important.
+    robot_body_fixture_def.restitution = 0.5;
+    robot_body_fixture_def.friction    = 0.0;
+    robot_body_fixture_def.userData =
+        new PhysicsObjectUserData({PhysicsObjectType::ROBOT_BODY, this});
 
-    robot_body = world->CreateBody(&robot_body_def);
+    b2PolygonShape* main_body_shape = getMainRobotBodyShape(robot, total_chicker_depth);
+    b2PolygonShape* front_left_body_shape =
+        getRobotBodyShapeFrontLeft(robot, total_chicker_depth);
+    b2PolygonShape* front_right_body_shape =
+        getRobotBodyShapeFrontRight(robot, total_chicker_depth);
+
+    auto body_shapes = {main_body_shape, front_left_body_shape, front_right_body_shape};
+    double total_shape_area = 0.0;
+    for (const auto* shape : body_shapes)
+    {
+        total_shape_area += polygonArea(*shape);
+    }
+    robot_body_fixture_def.density = mass_kg / total_shape_area;
+
+    for (const auto shape : body_shapes)
+    {
+        robot_body_fixture_def.shape = shape;
+        robot_body->CreateFixture(&robot_body_fixture_def);
+    }
+}
+
+void PhysicsRobot::setupDribblerFixture(const Robot& robot, double dribbler_depth)
+{
+    b2FixtureDef robot_dribbler_fixture_def;
+    // The dribbler has no mass in simulation. Mass is already accounted for by the robot
+    // body
+    robot_dribbler_fixture_def.density = 0.0;
```

Comment:
nit: can we put this number in the header file

Reply:
done